src/eth/EthereumService.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2019 ShipChain, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export interface TransactionEventHandlers {
+    receipt?: Function;
+    confirmation?: Function;
+    error?: Function;
+}
+
+export interface DeployedContractResult {
+    address;
+    author;
+    hash;
+}
+
+export abstract class EthereumService {
+    protected transactionConfirmations: number = 12;
```

Comment:
Maybe this should be a setting? How does this behave in an env like DEV where we typically only get 1 confirmation in a reasonable amount of time?

Reply:
This is only the default, the implemetation `EthersEthereumService` overrides with `transactionConfirmations = 1` when we are not running against a known network I.E. in DEV.